### ISSUE TITLE
layer-shell view steals the focus from scale plugin

### ISSUE THREAD

**Issue by dkondor** [WayfireWM/wayfire#2422]
**Describe the bug**
If scale is started when a layer-shell surface has the keyboard focus, it cannot grab the focus for itself.

**To Reproduce**
1. Enable the scale plugin
2. Ensure that a layer-shell surface has the keyboard focus
3. Start the scale plugin (keybinding or IPC works)

**Expected behavior**
Scale has keyboard interaction (e.g. arrows, ESC).

**Actual behavior**
Scale starts, but the layer-shell surface continues to have the keyboard focus. Using the mouse to interact with scale does work. Interestingly, the scale-title-filter plugin also works.

**Wayfire version**
git, 44e1fa9c62e1f8c9f35cedbf3e86c6a0247e0b79

This can be easily reproduced using the gtk-layer-demo program (setting keyboard interactivity to "On demand"). I'm also attaching a simple test case (in Python, but Github does not allow .py) that calls scale via IPC when a button is clicked.

Real-world case: [Cairo-Dock](https://github.com/Cairo-Dock/cairo-dock-core) with [this](https://github.com/dkondor/wayfire-scale-ipc) plugin

[ipc_window.txt](https://github.com/user-attachments/files/16489086/ipc_window.txt)

**Comment by ammen99** [comment]
Believe it or not, this is intentional. Scale's input grab is below the layer where panels and docks usually are. This is done so that you can indeed interact with wf-panel and other panels while in scale, someone requested this feature awhile ago. The solution/expectation is to have your dock not be keyboard interactive unless it really needs to interact with the keyboard.

**Comment by dkondor** [comment]
Thanks for the explanation! BTW is there a recommended way for "losing" the keyboard focus from a layer-shell view while keeping the availability to be focused later? Switching keyboard-interactivity to NONE and back to ON_DEMAND grabs again the keyboard focus.

**Comment by ammen99** [comment]
It shouldn't grab on the switch back to on-demand, this is probably a bug. On-demand should be auto-focused only on map.
